### PR TITLE
CinemaHall and MovieSession Dao and Service layers implemented

### PR DESCRIPTION
**Pull request by DmGrt**

### CODE REVIEW

**Review by RomanHreits — APPROVED** [review]
Good job!

File: src/main/java/com/dmgrt/cinema/models/MovieSession.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+@Entity
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
```
Comment (RomanDubovskyi) [inline src/main/java/com/dmgrt/cinema/models/MovieSession.java:19]: this is for your information:  keep in mind that toOne relations have fetchType eager by default, which is not always quite goo

File: src/main/java/com/dmgrt/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl extends AbstractDao<MovieSession> implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        LocalDateTime startOfDay = date.atStartOfDay();
+        LocalDateTime endOfDay = date.atTime(LocalTime.MAX);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dmgrt/cinema/dao/impl/MovieSessionDaoImpl.java]: these variables are redundant here, just pass values to setParameter method right away

File: src/main/java/com/dmgrt/cinema/models/MovieSession.java (hunk 1 above)
Comment (DmGrt) [reply src/main/java/com/dmgrt/cinema/models/MovieSession.java:19]: thanks

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**